cloudinit/sources/__init__.py

```diff
@@ -349,6 +354,18 @@ def _unpickle(self, ci_pkl_version: int) -> None:
                 raise DatasourceUnpickleUserDataError() from e
         if not hasattr(self, "extra_hotplug_udev_rules"):
             self.extra_hotplug_udev_rules = None
+        if not hasattr(self, "metadata_address"):
+            self.metadata_address = None
+        if not hasattr(self, "_platform_type"):
```

Comment:
I think we can drop the hasattr check on _platform_type in DataSourceNoCloud.py and Ec2 too